src/main/java/com/socrata/api/Soda2Producer.java

```diff
@@ -520,14 +520,20 @@ UpsertResult deserializeUpsertResult(InputStream is) throws IOException
                 currToken = parser.nextToken();
             }
 
-            return new UpsertResult(inserts, updates, deletes, errors.size() > 0 ? errors : null);
+            return new UpsertResult(inserts, updates, deletes, errors.size() > 0 ? errors : null, truthDataVersion);
         }
 
         return parser.readValueAs(UpsertResult.class);
     }
 
     UpsertResult deserializeUpsertResult(Response response) throws IOException {
-        return deserializeUpsertResult(response.readEntity(InputStream.class));
+        Long truthDataVersion = null;
+        try {
+            truthDataVersion = Long.parseLong(response.getHeaders().getFirst("X-SODA2-Truth-Version").toString());
```

Comment:
This header might not exist (e.g., if the upsert is being done on an OBE dataset) in which case you'll get a nullpointerexception here